qiskit/circuit/quantumcircuit.py

```diff
@@ -3889,15 +3889,15 @@ def clear(self) -> None:
     def _create_creg(self, length: int, name: str) -> ClassicalRegister:
         """Creates a creg, checking if ClassicalRegister with same name exists"""
         if name in [creg.name for creg in self.cregs]:
-            new_creg = ClassicalRegister(length, name=f"{name}{ClassicalRegister.instance_count}")
```

Comment:
It looks to like there is a behavior change here, in that the old version doesn't increment the instance count, but the new version does. Is that a concern?

Reply:
Ah, scratch that, I see now that this was happening before in `new`.

Reply:
Or actually (again lol), we're calling `py_new` from `_new_with_prefix` but also incrementing the instance count before we call it. So I _think_ that would double-increment, unless I'm missing something.

Reply:
1. The behavior change was actually in #13860 where we just retreived the counter and never incremented it, whereas before it would be incremented.
2. Calling `py_new` only increments the counter if no name is provided. Since we're already parsing the name beforehand, it should only count as one increment.

Reply:
Ah okay, I missed the subtly to point 2. 🙂